pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1505,6 +1531,53 @@ func (r *ReconcileClusterDeployment) cleanupInstallLogPVC(cd *hivev1.ClusterDepl
 	return nil
 }
 
+// deleteInstallPods deletes the install pods for the given provision
+func (r *ReconcileClusterDeployment) deleteInstallPods(provision *hivev1.ClusterProvision, pLog log.FieldLogger) error {
+	// fetching install job name for the provision and using this as label selector to fetch install pods
+	installJobName := install.GetInstallJobName(provision)
+	requirement, err := labels.NewRequirement("job-name", selection.Equals, []string{installJobName})
+
+	if err != nil {
+		pLog.WithError(err).Error("unexpected error generating label selector to fetch install pods")
+		return err
+	}
+	labelSelector := labels.NewSelector().Add(*requirement)
+	listOptions := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+
+	podList := &corev1.PodList{}
+	if err := r.List(context.TODO(), podList, listOptions); err != nil {
```

Comment:
For reference, this can be made simpler.
```suggestion
	podList := &corev1.PodList{}
	if err := r.List(
		context.TODO(),
		podList,
		client.InNamespace(provision.Namespace),
		client.MatchingLabels{"job-name": install.GetInstallJobName(provision)},
	); err != nil {
```